### PR TITLE
Additional highstate output modes; changes_id, filter_id, terse_id, full_id

### PR DESCRIPTION
**Pull request by The-Loeki**
### What does this PR do?

Cleanup the highstate outputter a bit and add support for `_id` to all output modes

### What issues does this PR fix or reference?
 #35137

### Previous Behavior
``mixed_id`` was added a while back and I personally fell in love with it's output. Except I want to see the changes in full and `changes_id` isn't there. 

Looking at the code it could use a little cleanup anyway, easily adding it as universal modifier along the way.

### CODE REVIEW

**Review by cachedout — CHANGES_REQUESTED** [review]
Eh? Why was the Windows minion config file removed here?

**Comment by The-Loeki** [comment]
@cachedout that was actually a merge conflict with @twangboy 's https://github.com/saltstack/salt/commit/7da86b3540876f1ef4646b6b8820ea8d4c5b97e8 ;)

I've adjusted the commits to the new situation better.

File: salt/output/highstate.py [hunk 1]
```diff
@@ -275,28 +285,17 @@ def _format_host(host, data):
                     continue
                 if str(ret['result']) in exclude:
                     continue
-            elif __opts__.get('state_output', 'full').lower() == 'terse':
-                # Print this chunk in a terse way and continue in the
-                # loop
+
+            elif any((
```
Comment (cachedout) [inline salt/output/highstate.py]: I'm guessing this is a part of the cleanup you mentioned? If so, please break it out into a separate PR. When we mix refactoring and feature work in a single PR it becomes very difficult to manage.

**Review by cachedout — CHANGES_REQUESTED** [review]

**Comment by The-Loeki** [comment]
Well, the way I refactored the code the additional _id options are implied because of [L251](https://github.com/saltstack/salt/pull/42063/files#diff-6783986d5efc4c6d746f4591ae8c1aabR251), leaving the feature PR w/a single silly line change and the documentation fixes.
This is why I chose this route; of course I'm willing to split it up in 2 PR's if you insist, 

PS. Glad to see you're still here ;) I hope the Nitrogen release hasn't cost you too much gray hair ;)

**Comment by rallytime** [comment]
Hi @The-Loeki - Apologies for the delay on this, but I just noticed that this is submitted against the `2017.7` branch, but since this is a feature addition, it needs to be submitted against the `develop` branch for the `Oxygen` release. Can you move this change over?

I agree with @cachedout that this should be broken up into 2 PRs as well.

Thank you very much for your contributions!

**Comment by The-Loeki** [comment]
NP, see #43501 & #43503

**Pull request closed without merging**